src/components/withNavigation.tsx

```diff
@@ -0,0 +1,23 @@
+import React, {ComponentType, ForwardedRef, RefAttributes} from 'react';
+import {NavigationProp, useNavigation} from '@react-navigation/native';
+import getComponentDisplayName from '../libs/getComponentDisplayName';
+
+type WithNavigationProps = {
+    navigation: NavigationProp<ReactNavigation.RootParamList>;
+};
+
+export default function withNavigation<TProps extends WithNavigationProps, TRef>(WrappedComponent: ComponentType<TProps & RefAttributes<TRef>>) {
+    function WithNavigation(props: Omit<TProps, keyof WithNavigationProps>, ref: ForwardedRef<TRef>) {
+        const navigation = useNavigation();
+        return (
+            <WrappedComponent
+                {...(props as TProps)}
+                ref={ref}
+                navigation={navigation}
+            />
+        );
+    }
+
+    WithNavigation.displayName = `withNavigation(${getComponentDisplayName(WrappedComponent as ComponentType)})`;
```

Comment:
@JKobrynski Maybe you can add [same update](https://github.com/Expensify/App/pull/29284/files#diff-3401552ecd81836fbf3b40788fa80901ec80b064b37a99c75ead8cf1e38384deR4) to `getComponentDisplayName` to get rid of the type assertion here

Reply:
Done!